27_tictactoe/lib/board.py

```diff
@@ -0,0 +1,79 @@
+from lib.player import Player
+from random import randint
+
+"""
+start(p1, p2) - starts a new game with players p1 and p2
+
+play(row,column) - current player picked a square. If the square was free mark it as X or O and switch turns to the other player.
+If square was taken or not in the board, or game is over, raise an exception.
+
+winner() - If game is over return the winning player, else return None
+
+val(row, column) - return current value for square (X, O or empty)
+"""
+
+class InvalidMove(Exception):
+    def __init__(self,msg):
+        self.message = "Invalid move in a tick-tack-toe; %s" %(msg)
+
+
+class Game:
+    def __init__(self):
+        self.reset_board()
+        self.players = [None,None]
+        self.current_player = None
+        self.winner = None
+        self.moves_counter = 0
+    
+    def reset_board(self):
+        self.data = [ [None,None,None] , 

```

Comment:
Python has * operator for lists, so this also works: `self.data = [[None] * 3] * 3`